src/generator/pollers.ts

```diff
@@ -49,25 +49,711 @@ type ${poller.name} struct {
 	client *${poller.client}
 	// polling tracker
-	// pt *pollingTracker${pascalCase(poller.pollingMethod)}
+	pt *pollingTracker${pascalCase(poller.pollingMethod)}
 }
 
+// Done returns true if the polling operation has terminated either in a success case or failure case,
+// otherwise it will return false
 func (p *${poller.name}) Done() bool {
-	return false
+	return p.pt.hasTerminated()
 }
 
+// ID ... 
 func (p *${poller.name}) ID() string {
-	return "NYI"
+	return ""
 }
 
 func (p *${poller.name}) Poll(ctx context.Context) (*${responseType}, error) {
-	return nil, nil
+	done, err := p.doneWithContext(ctx)
+	if done {
+		resp := p.response()
+		result, err := p.client.${poller.operationName}HandleResponse(resp)
+		if err != nil {
+			return nil, err
+		}
+		return result, nil
+	}
+	return nil, err
 }
 
 func (p *${poller.name}) Wait(ctx context.Context, pollingInterval time.Duration) (*${responseType}, error) {
 	return nil, nil
 }
+
+// Response returns the last HTTP response.
+func (p *${poller.name}) response() *azcore.Response {
+	if p.pt == nil {
+		return nil
+	}
+	return p.pt.latestResponse()
+}
+
+// doneWithContext queries the service to see if the operation has completed.
+func (p *${poller.name}) doneWithContext(ctx context.Context) (done bool, err error) {
+	if p.pt == nil {
+		return false, errors.New("poller is not initialized")
+	}
+	if p.pt.hasTerminated() {
+		return true, p.pt.pollingError()
+	}
+	if err := p.pt.pollForStatus(ctx, p.client.p); err != nil {
+		return false, err
+	}
+	if err := p.pt.checkForErrors(); err != nil {
+		return p.pt.hasTerminated(), err
+	}
+	if err := p.pt.updatePollingState(p.pt.provisioningStateApplicable()); err != nil {
+		return false, err
+	}
+	if err := p.pt.initPollingMethod(); err != nil {
+		return false, err
+	}
+	if err := p.pt.updatePollingMethod(); err != nil {
+		return false, err
+	}
+	return p.pt.hasTerminated(), p.pt.pollingError()
+}
+
 `;
   }
   return text;
 }
+
+// Creates the content in pollers_helper.go
+export async function generatePollersHelper(session: Session<CodeModel>): Promise<string> {
+	if (session.model.language.go!.pollerTypes === undefined) {
+	  return '';
+	}
+	let text = await contentPreamble(session);
+	const pollers = <Array<PollerInfo>>session.model.language.go!.pollerTypes;
+  	pollers.sort((a: PollerInfo, b: PollerInfo) => { return sortAscending(a.name, b.name) });
+  
+	// add standard imports
+	const imports = new ImportManager();
+	imports.add('bytes');
+	imports.add('context');
+	imports.add('encoding/json');
+	imports.add('errors');
+	imports.add('github.com/Azure/azure-sdk-for-go/sdk/azcore');
+	imports.add('io/ioutil');
+	imports.add('net/http');
+	imports.add('net/url');
+	imports.add('strings');
+	imports.add('fmt');
+	imports.add('errors');
+	text += imports.text();
+	let errorCtor = pollers[0].pollingError.language.go!.constructorName;
+	// TODO separate this into manageable chunks of text by section of functionality
```

Comment:
This will be included in a future PR

Reply:
Do you mean splitting this up into multiple helper files?

Reply:
No no, I simply want to split up this huge chunk of text into smaller named vars. Like constDeclarations, pollerBaseDefinition, pollertrackerdeleteMethods, things like that, that could make it more manageable to make changes to this code. What do you think? That's what I came up with to make this easier to modify in the future.

Reply:
I see what you mean now.  What would be great is if we had a way to have templated Go code (i.e. actual `.go` source files) that we could consume as part of code generation.  Kind of like what we had in track 1 with the razor files, however we only use them for things like this where the majority of the code is constant (was thinking about this when I did pagers).